### ISSUE TITLE
Multiple Textures not working! 

### ISSUE THREAD

**Issue by JeannotLandry** [bsavery/PRMan-for-Blender#128]
I designed a little mascot for the fun of it and to test out PRman for Blender in terms of character work.

There is currently two textures in the scene. One for the skin and one for the iris of the eyeball (separated objects) and the skin isn't showing and shows the Iris texture on the skin instead. Everything seems to be plugged in well in the nodes. Not sure if I am doing anything wrong. The only way I can get the skin texture to show up is to delete the nodes for the iris material and save the scene and re-open. Problem solved, only until I redo the iris material, than issue pops back again. It seems that whatever texture I load in First will be the one showing in all the next materials I assign textures to.

If I load the skin texture first, that also appears on the iris instead of the iris, and vice versa! It also seems to overwrite the HDRI image I have set on the environment light!

Here is an image of the issue!
![rendersissuecombined](https://cloud.githubusercontent.com/assets/14031486/9606414/b51c2b82-5099-11e5-8967-3e3ae45a78e6.jpg)
![blenderman conception1](https://cloud.githubusercontent.com/assets/14031486/9606423/bed54bcc-5099-11e5-9b2d-d19f49d71835.jpg)

**Comment by JeannotLandry** [comment]
I've uploaded the .blend in the forum:
https://renderman.pixar.com/forum/showthread.php?s=&threadid=28956

**Comment by MWY3510** [comment]
Well I thought that this matter was more pressing (since it is a bug.) I found no problem with multiple textures on my end. Even using the provided scene I was able to render everything without problems.
![mascot](https://cloud.githubusercontent.com/assets/9410101/9648300/2e1332c6-51b5-11e5-91ed-64ac558586d4.png)

**Comment by bsavery** [comment]
MWY.  I think this is a sporadic crash related to texture params.  I was trying to track down one in another place and was looking at the output of gen_params.  I noticed the params had a "enum filename": 'periodic'

Might be a clue.

**Comment by bsavery** [comment]
particularly it looks like in shader_parameters.py we add some texture options, particularly get a second filename...  those get emitted in the rib as well it look like, which is bad.

**Comment by bsavery** [comment]
Made a quick and dirty fix for above issue.

**Comment by MWY3510** [comment]
Sorry about that. I thought I tracked down everything, I guess I was mistaken. I'll take a look sometime soon to see if there are more issues with that.

**Comment by bsavery** [comment]
Yeah, just take a look at what gen_params puts into the rib stream for the
texture nodes.  No worries.

On Thu, Sep 3, 2015 at 5:03 PM, MWY3510 <notifications@github.com> wrote:

> Sorry about that. I thought I tracked down everything, I guess I was
> mistaken. I'll take a look sometime soon to see if there are more issues
> with that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/issues/128#issuecomment-137605069>
> .
>



-- 
brian.savery@gmail.com
508-274-8700

**Comment by bsavery** [comment]
Were the issues fixed with the latest checkin?  Jeannot?

**Comment by JeannotLandry** [comment]
bsavery

Sorry for the lack of being here, got sick after taking care of my wife and then my son got sick too. Lol, crazy past few days. Well, I downloaded and installed the last version from github and now as soon as I open the character scene, it closes right back. No worries though, I did export an obj to use in ZBrush, but for some reasons my ZBrush does funky things. I am starting to think my computer either needs and upgrade or a good clean up.

I am currently running Win 7 on an HP AMD Phenome II x6 2.9 Ghz over clocked at 3.4, 16 Gigs Ram and a AMD Radeon 7800 2 Gigs. Computer is a little old, almost 5 years, but has never let me down thus far. I am running on dual display with my Wacom HD tablet.

I am thinking of setting it up for Cycles and giving you the files so you can test them with PRman, my lack of understanding PRman is also a possible issue...

**Comment by JeannotLandry** [comment]
Hey bsavery,

Sorry for the late reply, just downloaded the latest version and gave it a go, everything works fine, multiple textures and IT renders again.

So since it works I think its safe to say we can close this issue.

**Comment by JeannotLandry** [comment]
Well, now the displacement works but it takes over the other textures. So it isn't fixed.

See attached image.
![mascot_disptakingovertextures](https://cloud.githubusercontent.com/assets/14031486/9828191/63230c72-58c0-11e5-93fd-9666b7b11e11.jpg)

**Comment by MWY3510** [comment]
Can we get a picture of the node set-up?

**Comment by JeannotLandry** [comment]
Sorry, I fired my computer a few days ago the same day I posted that (didn't fried it with renderman, I was just sketching in sketchbook when it happened) and I just received my new computer yesterday. I have lost that work since I didn't backed it up. I may be able to extract it from the HDD but I am not 100% sure of the extent of the damages since it sorta smoked lol. I'll have to check that up and at worst, I'll either recreate it or create something entirely different.

**Comment by bsavery** [comment]
So this is fixed now right?

**Comment by JeannotLandry** [comment]
I am not sure, I'll let you know as soon as I create textures and displacement for this little dude, right now its only materials. I did notice something strange though. When I was reposing the character, it didn't carry over. To make it work, I had to hide the subsurf modifier and then it worked. Also, when I have the hair material selected, it carries over that material to the skin, so I need to make sure I select the skin material for it to work. Just a little weird, but they both have workarounds.

![alienduderendermantest001](https://cloud.githubusercontent.com/assets/14031486/11644377/e38ef06a-9d21-11e5-9595-54c2f7ad692b.jpg)

**Comment by JeannotLandry** [comment]
I do have the latest version installed in case you ask, and I will let you know as soon as I put textures on that little dude!

**Comment by JeannotLandry** [comment]
Well, multiple textures seemed to work fine. Until I tried to use the export UV for hair and plugged a texture for random strand color and since then, I cannot render at all using Renderman with IT or if I press render button, It says empty! Now if I create a new scene, everything works fine, but If I open up my character scene, it doesn't work! So I tried installing the newest plugin version, disabled it in my scene, re created all the materials and nodes, and still, that scene wont render anymore since I have used that export mesh UV in the particle tab! Any clues on what I can do to fix this?
